libparsec/crates/types/src/certif.rs

```diff
@@ -48,10 +48,12 @@ fn dump<T>(obj: &T) -> Vec<u8>
 where
     T: Serialize,
 {
-    let serialized = ::rmp_serde::to_vec_named(obj).unwrap_or_else(|_| unreachable!());
+    let serialized = ::rmp_serde::to_vec_named(obj).expect("object should be serializable");
     let mut e = ZlibEncoder::new(Vec::new(), flate2::Compression::default());
-    e.write_all(&serialized).unwrap_or_else(|_| unreachable!());
-    e.finish().unwrap_or_else(|_| unreachable!())
+    e.write_all(&serialized)
+        .expect("serialized object should be writable to ZlibEncoder");
+    e.finish()
+        .expect("flushing ZlibEncoder output stream should not generate I/O error")
```

Comment:
Could be more precise here, we won't have `I/O` error because the buffer is in memory (but we could still get an OOM but that something else :wink:)

Reply:
Are you suggesting something like `"in-memory buffer should not fail"` in both cases here?

Reply:
yes